test/unit/org/apache/cassandra/db/compaction/LeveledCompactionStrategyTest.java

```
@@ -969,6 +969,36 @@ public void testReduceScopeL0L1() throws IOException
         }
     }
 
+    @Test()
+    public void testInvalidFanoutAndSSTableSize() {
+        try {
+            Map<String, String> options = new HashMap<>();
+            options.put("class", "LeveledCompactionStrategy");
+            options.put("fanout_size", "90");
+            options.put("sstable_size_in_mb", "1089");
+            LeveledCompactionStrategy.validateOptions(options);
+        }
+        catch (ConfigurationException e) {
+            assertTrue(e.getMessage().contains("your maxSSTableSize must be absurdly high to compute"));
+            return;
+        }
+        Assert.fail("fanout_sizeed and sstable_size_in_mb are invalid, but did not throw ConfigurationException");
+    }
+
+    @Test
+    public void testValidOptions() {
+        try {
+            Map<String, String> options = new HashMap<>();
+            options.put("class", "LeveledCompactionStrategy");
+            options.put("fanout_size", "10");
+            options.put("sstable_size_in_mb", "160");
+            LeveledCompactionStrategy.validateOptions(options);
+        }
+        catch (Exception e) {
+            Assert.fail("fanout_sizeed and sstable_size_in_mb are valid, but threw  " + e.getClass().getName());
```

can we remove the try/catch?  its cleaner to just let it bubble up than to recatch and loose context.  It also is harder to read as I have to now understand why we are catching

---

Done